src/main/java/Main.java

```diff
@@ -0,0 +1,36 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
```

Comment:
Leave this class (Main) on current level.
All other supporting classes move into deeper packages (sub-folders).


Additionally, I would simplify whole application in following way:
Main should only contain psvm, and in int one class injected - "Processor". With it id indicates which file to process, 

*******************************************

Structure:

package:java
MAIN 
         |
         package.logic
         |Processor (reads, process, writes - it indicates name of output by it self.)
                         package.support
                         |FileHelper (read/write)
                         |NumbersProcessor(Process single String)

Additionally, add tests with mocking.
Remember about dependency Injection.
I.e. your class Main should have it written in following way :

*******************************************
public class Main{
    private Processor processor;   <-----this class

    public Main(Processor processor){ <--is injected here
        this.processor = processor;
    }
*******************************************